bridge-token-factory/src/lib.rs

```diff
@@ -92,6 +96,20 @@ pub trait ExtBridgeTokenFactory {
         #[serializer(borsh)] amount: Balance,
         #[serializer(borsh)] proof: Proof,
     ) -> Promise;
+
+    #[result_serializer(borsh)]
+    fn finish_updating_metadata(
+        &mut self,
+        #[callback]
+        #[serializer(borsh)]
+        verification_success: bool,
+        #[serializer(borsh)] token: String,
+        #[serializer(borsh)] name: String,
+        #[serializer(borsh)] symbol: String,
+        #[serializer(borsh)] decimals: u8,
+        #[serializer(borsh)] timestamp: u64,
+        #[serializer(borsh)] proof: Proof,
+    ) -> Promise;
 }
```

Comment:
As the metadata could be updated either by calling either `set_metadata` directly or by verifying the proof from the `MetadataConnector`, I suggest renaming the method to something like:  `finish_updating_metadata_with_proof` to be explicit on its purpose and to eliminate possible misleading.